address_lookup_tables/src/address_lookup_table_store.rs

```diff
@@ -0,0 +1,218 @@
+use async_trait::async_trait;
+use dashmap::DashMap;
+use itertools::Itertools;
+use prometheus::{opts, register_int_gauge, IntGauge};
+use serde::{Deserialize, Serialize};
+use solana_address_lookup_table_program::state::AddressLookupTable;
+use solana_lite_rpc_core::traits::address_lookup_table_interface::AddressLookupTableInterface;
+use solana_rpc_client::nonblocking::rpc_client::RpcClient;
+use solana_sdk::{commitment_config::CommitmentConfig, pubkey::Pubkey};
+use std::{sync::Arc, time::Duration};
+
+lazy_static::lazy_static! {
+    static ref LRPC_ALTS_IN_STORE: IntGauge =
+       register_int_gauge!(opts!("literpc_alts_stored", "Alts stored in literpc")).unwrap();
+}
+
+#[derive(Clone)]
+pub struct AddressLookupTableStore {
+    rpc_client: Arc<RpcClient>,
+    pub map: Arc<DashMap<Pubkey, Vec<Pubkey>>>,
+}
+
+impl AddressLookupTableStore {
+    pub fn new(rpc_client: Arc<RpcClient>) -> Self {
+        Self {
+            rpc_client,
+            map: Arc::new(DashMap::new()),
+        }
+    }
+
+    async fn load_alts_list(&self, alts_list: &[Pubkey]) {
+        log::trace!("Preloading {} ALTs", alts_list.len());
+        for batches in alts_list.chunks(1000).map(|x| x.to_vec()) {
+            let tasks = batches.chunks(100).map(|batch| {
+                let batch = batch.to_vec();
+                let rpc_client = self.rpc_client.clone();
+                let this = self.clone();
+                tokio::spawn(async move {
+                    let data = rpc_client
+                        .get_multiple_accounts_with_commitment(
+                            &batch,
+                            CommitmentConfig::processed(),
+                        )
+                        .await;
+
+                    match data {
+                        Ok(multiple_accounts) => {
+                            for (index, acc) in multiple_accounts.value.iter().enumerate() {
+                                if let Some(acc) = acc {
+                                    this.save_account(&batch[index], &acc.data);
+                                }
+                            }
+                        }
+                        Err(e) => {
+                            log::error!(
+                                "error loading {} alts with error {}",
```

Comment:
pls write in log, what you do in case of error i.e. "skipping lookup for batch N and continue"